### PR TITLE
Add sample code for avc decode streamout

### PR DESCRIPTION
**Pull request by Jexu**
Add new folder 'vendor/intel' and give sample code for how to use avc streamout

Signed-off-by: Zhengguo.Xu <zhengguo.xu@intel.com>

### CODE REVIEW

File: vendor/intel/avcstreamoutdemo.c [hunk 1]
```diff
@@ -0,0 +1,1436 @@
...
+      direct_spatial_mv_pred_flag:0,
+      num_ref_idx_l0_active_minus1:0,
+      num_ref_idx_l1_active_minus1:0,
+      cabac_init_idc:0,
+      slice_qp_delta:-1,
+      disable_deblocking_filter_idc:0,
+      slice_alpha_c0_offset_div2:0,
+      slice_beta_offset_div2:0,
+      RefPicList0:{
+          {4294967295, 0, 1, 0, 0},
```
Comment (XinfengZhang) [inline vendor/intel/avcstreamoutdemo.c]: it is 0xffffffff, invalidate picture, right?

**Comment by XinfengZhang** [comment]
LGTM, but if possible, add more description comment in the code

File: vendor/intel/avcstreamoutdemo.c (hunk 1 above)
Comment (Jexu) [reply vendor/intel/avcstreamoutdemo.c]: @XinfengZhang yes, it is

File: vendor/intel/avcstreamoutdemo.c [hunk 2]
```diff
@@ -0,0 +1,1436 @@
...
+
+/* hardcoded here without a bitstream parser helper
+ */
+static VAPictureParameterBufferH264 pic_param[2] ={
+  {
+      CurrPic:{
+          0, 0, 8, 0, 0
+      },
+      ReferenceFrames:{
+          {4294967295, 0, 1, 0, 0},
```
Comment (XinfengZhang) [inline vendor/intel/avcstreamoutdemo.c]: use VA_PICTURE_H264_INVALID, VA_INVALID_SURFACE, these definition in va is more suitable than the constant

**Comment by XinfengZhang** [comment]
please help to check these warning:
my gcc version is 4.8.5
avcstreamoutdemo.h:218:8: warning: extra tokens at end of #endif directive [enabled by default]
 #endif _AVC_STREAMOUT_DEMO_H_
        ^
avcstreamoutdemo.c:553:13: warning: missing braces around initializer [-Wmissing-braces]
             0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10,
             ^
avcstreamoutdemo.c:553:13: warning: (near initialization for ‘iq_matrix[0].ScalingList4x4[0]’) [-Wmissing-braces]
  CCLD     avcstreamoutdemo

**Pull request merged**